sdk/typescript/introspector/scanner/utils.ts

```diff
@@ -182,9 +183,16 @@ export function isOptional(param: ts.Symbol): OptionalValue {
 
     // Convert the symbol declaration into Parameter
     if (ts.isParameter(parameterDeclaration)) {
-      result.optional =
-        parameterDeclaration.questionToken !== undefined ||
-        parameterDeclaration.initializer !== undefined
+      // Check for ? notation
+      result.optional = parameterDeclaration.questionToken !== undefined
+
+      // Check for `<xx>|null` notation
+      if (parameterDeclaration.type) {
+        const type = parameterDeclaration.type.getText()
+        if (type.includes("null")) {
```

Comment:
How would you represent this?
```graphql
foo: [String]!
```
Maybe:
```typescript
(string | null)[]
```
In that case, this would make the list nullable, when it should be non-null.

Reply:
Fixed in [55606ed](https://github.com/dagger/dagger/pull/6766/commits/55606edc6b117cee7a1b579e4ea9fe9d361671ca)